src/modules/booking/components/service-booking-modal/service-booking-modal.tsx

```diff
@@ -0,0 +1,133 @@
+import { type FC, useState } from 'react';
+// components
+import { RadioButton } from '@/modules/core/components/radio-button';
+import { Avatar } from '@/modules/core/components/avatar';
+import { Button } from '@/modules/core/components/button';
+import { Stepper } from '@/modules/core/components/stepper';
+import { Dialog } from '@/modules/core/components/dialog';
+import { Typography } from '@/modules/core/components/typogrpahy';
+import { BookingForm } from '@/modules/booking/components/booking-form';
+import { BookingCalendar } from '@/modules/booking/components/booking-calendar';
+import { BaseCardWithRadioButton } from '../booking-card-radio-button';
+// type
+import { type ModalProps } from '@/modules/core/components/dialog/dialog.interface';
+// utils
+import { trpc } from '@/modules/core/utils/trpc.utils';
+
+import styles from './service-booking-modal.module.scss';
+
+export const ServiceBookingModal: FC<Omit<ModalProps, 'children'>> = (
+  props
+) => {
+  // state
+  const [value, setValue] = useState<string>('');
+
+  const [step, setStep] = useState<'service' | 'datetime' | 'confirmation'>(
+    'service'
+  );
+
+  const handleNext = () => {
+    if (step === 'service') {
+      setStep('datetime');
+    } else if (step === 'datetime') {
+      setStep('confirmation');
+    }
+  };
+
+  const handleBack = () => {
+    if (step === 'confirmation') {
+      setStep('datetime');
+    } else if (step === 'datetime') {
+      setStep('service');
+    }
+  };
+
+  const { data: serviceList } = trpc.serviceOnProfessional.list.useQuery({
+    limit: 10,
+    offset: 0,
+  });
```

Comment:
update this part to:

```
const { data: me } = trpc.user.me.useQuery({ expand: ['professional' ]});
const { data: serviceList } = trpc.serviceOnProfessional.list.useQuery({
    limit: 10,
    offset: 0,
    professionalId: me?.professional?.id,
  }, {
    enabled: Boolean(me?.professional?.id)
  });
```

This is needed to request only services that was created by you